shared/functions/sql/leagues_repository.py

```
@@ -44,20 +33,28 @@ def read_dataframe(self) -> pd.DataFrame:
+        # Assign the current timestamp to all rows
+        df["creation_timestamp"] = current_version
+
         try:
-            leagues = [LeagueOverview(**row) for row in df.to_dict('records')]
-            session.bulk_save_objects(leagues)
+            # Insert new data with a new version
+            session.bulk_save_objects([LeagueOverview(**row) for row in df.to_dict('records')])
             session.commit()
+
+            # Once successful, delete older versions
+            session.execute(text(f"DELETE FROM league_overview WHERE creation_timestamp < {current_version}"))
```

Ruff (0.14.0)

Possible SQL injection vector through string-based query construction

(S608)